### ISSUE TITLE
peptides

### ISSUE THREAD

**Issue by ZhuNing9527** [chaidiscovery/chai-lab#149]
How to predict the structure of cyclic  peptides (amide bonds forming a ring)

**Comment by aozalevsky** [comment]
You can submit it as a ligand using SMILES. 

P.S. Not a ChaiDiscovery employee; was just passing by.

**Comment by arogozhnikov** [comment]
@ZhuNing9527
Idea for you to try: add a max distance restraint between first and last residue of peptide

**Comment by ZhuNing9527** [comment]
Thank you for your suggestion. I am trying it. In addition to this, I wonder if I can modify the relative position encoding method to achieve

**Comment by ZhuNing9527** [comment]
> @ZhuNing9527 Idea for you to try: add a max distance restraint between first and last residue of peptide

Thank you for your suggestion. I am trying it. In addition to this, I wonder if I can modify the relative position encoding method to achieve

**Comment by arogozhnikov** [comment]
@ZhuNing9527 

> I wonder if I can modify the relative position encoding method to achieve

Yup, you need to modify here: https://github.com/chaidiscovery/chai-lab/blob/f77f44165ef59bba0685f176d03ee4b02e793967/chai_lab/data/features/generators/relative_token.py#L36-L53
